### PR TITLE
additional questions for ML chapter 12

### PR DESCRIPTION
**Pull request by tvpeter**
- add more questions for Mastering Lightning Network Chapter 12.

### CODE REVIEW

File: mastering-lightning/README.md [hunk 1]
```diff
@@ -112,6 +112,8 @@ https://github.com/lnbook/lnbook
 1. Maintaining a channel graph and performing pathfinding are computationally quite expensive. Could we offload that work to third parties? How would that impact an individual's privacy, security and ability to receive payments?
 1. In which ways does pathfinding rely on messages from the gossip protocol?
 1. Could a malicious actor lie about his channel's fees? How would that affect the network, and how can we assume it's not in his best interest to do so?
+1. Does sending MPP incur more fees compare to sending a single payment if the `base_fee` and `fee_rate` were the same?
```
Comment (stickies-v) [inline mastering-lightning/README.md]: It's a good starting point, but having equal `base_fee` and `fee_rate` makes it a pretty straightforward/obvious question imo. The interesting part is where `base_fee` and `fee_rate` are not necessarily the same, and nodes can optimize routes. Would extend it a bit.
```suggestion
1. Does sending a Multi-Path Payment (MPP) incur more fees compared to sending a single payment if the `base_fee` and `fee_rate` are the same? Is 
```

File: mastering-lightning/README.md [hunk 2]
```diff
@@ -112,6 +112,8 @@ https://github.com/lnbook/lnbook
 1. Maintaining a channel graph and performing pathfinding are computationally quite expensive. Could we offload that work to third parties? How would that impact an individual's privacy, security and ability to receive payments?
 1. In which ways does pathfinding rely on messages from the gossip protocol?
 1. Could a malicious actor lie about his channel's fees? How would that affect the network, and how can we assume it's not in his best interest to do so?
+1. Does sending MPP incur more fees compare to sending a single payment if the `base_fee` and `fee_rate` were the same? 
+1. Are there any differences between MPP and Atomic MPP?
```
Comment (stickies-v) [inline mastering-lightning/README.md]: Suggestion to make it more a discussion question than a strictly factual question:
```suggestion
1. What are the differences between MPP and Atomic MPP? Is one strictly better than the other? What are the trade-offs?
```

File: mastering-lightning/README.md (hunk 1 above)
Comment (stickies-v) [reply mastering-lightning/README.md]: What do you think about extending the question? Unmarking as resolved.

(Also noticed an additional "Is" sneaked in at the end of my suggestion)

Comment (tvpeter) [reply mastering-lightning/README.md]: Yes, seen it. I want to add a part where they are not the same. I committed so I can get it to my local system. setting up a new system.

File: mastering-lightning/README.md [hunk 3]
```diff
@@ -112,6 +112,9 @@ https://github.com/lnbook/lnbook
 1. Maintaining a channel graph and performing pathfinding are computationally quite expensive. Could we offload that work to third parties? How would that impact an individual's privacy, security and ability to receive payments?
 1. In which ways does pathfinding rely on messages from the gossip protocol?
 1. Could a malicious actor lie about his channel's fees? How would that affect the network, and how can we assume it's not in his best interest to do so?
+1. Does sending a Multi-Path Payment (MPP) incur more fees compared to sending a single payment if the `base_fee` and `fee_rate` are the same? When will you consider one over the other?
+1. What are the differences between MPP and Atomic MPP? Is one strictly better than the other? What are the trade-offs?
+1. What happens when some parts of a Multi-Path Payment (MPP) succeeds and others fail?
```
Comment (stickies-v) [inline mastering-lightning/README.md]: I think this is  already covered by the previous question?
```suggestion
1. What happens when some parts of a Multi-Path Payment (MPP) succeed and others fail?
```

File: mastering-lightning/README.md [hunk 4]
```diff
@@ -112,6 +112,9 @@ https://github.com/lnbook/lnbook
 1. Maintaining a channel graph and performing pathfinding are computationally quite expensive. Could we offload that work to third parties? How would that impact an individual's privacy, security and ability to receive payments?
 1. In which ways does pathfinding rely on messages from the gossip protocol?
 1. Could a malicious actor lie about his channel's fees? How would that affect the network, and how can we assume it's not in his best interest to do so?
+1. Does sending a Multi-Path Payment (MPP) incur more fees compared to sending a single payment if the `base_fee` and `fee_rate` are the same? When will you consider one over the other?
```
Comment (stickies-v) [inline mastering-lightning/README.md]: This phrasing still implies that `base_fee` and `fee_rate` are the same throughout the question, limiting the range of answers.

```suggestion
1. Compare a Multi-Path Payment (MPP) with a single payment. Assuming that `base_fee` and `fee_rate` are the same: how do the total fees compare? Is the fee assumption reasonable? In real life (without artificial assumptions), when will you consider one over the other?
```

**Review by stickies-v — APPROVED** [review]
LGTM, thanks @tvpeter !

**Pull request merged**